src/PollingBlockTracker.ts

```diff
@@ -299,6 +345,15 @@ export class PollingBlockTracker
 
       try {
         this.emit('error', newErr);
+        if (
+          this.listeners('error').filter((listener) =>
+            this.#internalEventListeners.every(
+              (internalListener) => !Object.is(listener, internalListener),
+            ),
+          ).length === 0
+        ) {
+          console.error(newErr);
+        }
```

Comment:
The behavior of this try / catch block is unclear to me. It looks like we want to log the error when `this.emit('error', newErr);` throws, but looking at unit tests what we want is log the error when there is no listener attached to the `error` event.

Since we now have an internal listener on `error` as well, we need to exclude that

Reply:
Exactly, we override `emit` in `@metamask/safe-event-emitter` to throw if there are no listeners for `error`. If this does happen, we are catching this and using `console.error` to log the error instead.

I do not understand why `@metamask/safe-event-emitter` does this, and I do not understand why we do not always log the error anyway. I think this is a very old line.

This seems fine to maintain the current behavior.